### PR TITLE
Review FAQ django app [#LP-86]

### PR DESCRIPTION
**Pull request by rosheen33**
* Added a model to store title and content of a Custom FAQ Page
* Added migration file
* Custom Admin panel **FaqAdmin**
* Template file added for Custom FAQ page
* View added to show dynamic content of page

connects #LP-86

### CODE REVIEW

File: lms/djangoapps/faq/admin.py [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+    def disable(self, request, queryset):
+        queryset.update(is_active=False)
+
+    disable.short_description = "Disable the selected pages"
+
+    def has_delete_permission(self, request, obj=None):
+        return False
+
+    def save_model(self, request, obj, form, change):
+        obj.added_by = request.user
```
Comment (ziafazal) [inline lms/djangoapps/faq/admin.py]: why we are not tracking `changed_by`?

File: lms/djangoapps/faq/models.py [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+from student.models import User
+from django.utils import timezone
+
+
+class Faq(models.Model):
+    title = models.CharField(max_length=128, default="FAQ")
+    content = models.TextField()
+    created_at = models.DateTimeField(default=timezone.now)
+    updated_at = models.DateTimeField(auto_now=True)
+    added_by = models.ForeignKey(User, related_name='faq_user')
```
Comment (ziafazal) [inline lms/djangoapps/faq/models.py]: worth adding `updated_by` field here.

File: lms/djangoapps/faq/models.py [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+    is_active = models.BooleanField(default=True)
+
+    def __str__(self):
```
Comment (ziafazal) [inline lms/djangoapps/faq/models.py]: Replace this with 
```
def __unicode__(self):
    return u'{} | {}'.format(self.title, self.is_active)
```
to handle unicodes.

File: lms/templates/footer.html [hunk 4]
```diff
@@ -83,3 +87,15 @@
     <link rel="stylesheet" type="text/css" href="${url}"></link>
   % endfor
 % endif
+
+<script>
+    $(document).ready(function(){
+        $.ajax({
```
Comment (ziafazal) [inline lms/templates/footer.html]: It does not make sense to hit db every time we have to load a page just to fill in FAQ title. Please override `footer.html` in philu's theme and name FAQ link to whatever needed.

File: lms/templates/faq/custom_faq.html [hunk 5]
```diff
@@ -0,0 +1,12 @@
+## mako
+<%page expression_filter="h"/>
```
Comment (ziafazal) [inline lms/templates/faq/custom_faq.html]: This html template should be in philu's theme.

File: lms/djangoapps/faq/views.py [hunk 6]
```diff
@@ -0,0 +1,49 @@
...
+        context = {
+            'title': faq_page.title,
+            'body': mark_safe(body.group(1)) if body else 'No Content',
+        }
+
+    return render_to_response("faq/custom_faq.html", context)
+
+
+@csrf_exempt
+def get_faq_title(request):
```
Comment (ziafazal) [inline lms/djangoapps/faq/views.py]: This view is obsolete see comments below.

File: lms/djangoapps/faq/views.py [hunk 7]
```diff
@@ -0,0 +1,49 @@
...
+
+    faq_page = Faq.objects.filter(is_active=True).last()
+
+    if not faq_page:
+        context = {
+            'title': 'FAQ',
+            'body': '** Please add content for FAQ page',
+        }
+    else:
+        body = re.search('<body>(.*)</body>', faq_page.content, re.S)
```
Comment (ziafazal) [inline lms/djangoapps/faq/views.py]: why we are searching in body tags? why not strip them off if they exist?

File: lms/templates/footer.html (hunk 4 above)
Comment (rosheen33) [reply lms/templates/footer.html]: FAQ title can be changed through admin panel anytime by admin.

File: lms/djangoapps/faq/views.py (hunk 7 above)
Comment (rosheen33) [reply lms/djangoapps/faq/views.py]: We have a source of a complete html page at backend we just need to extract the body from it.

File: lms/templates/faq/custom_faq.html (hunk 5 above)
Comment (zee-pk) [reply lms/templates/faq/custom_faq.html]: We are setting up PhilU theme, from next sprint (which is starting tomorrow) we will make all template/style changes in PhilU theme.

File: lms/djangoapps/faq/models.py (hunk 2 above)
Comment (rosheen33) [reply lms/djangoapps/faq/models.py]: Done

File: lms/djangoapps/faq/models.py (hunk 3 above)
Comment (rosheen33) [reply lms/djangoapps/faq/models.py]: Done

File: lms/djangoapps/faq/admin.py (hunk 1 above)
Comment (rosheen33) [reply lms/djangoapps/faq/admin.py]: Done

**Comment by rosheen33** [comment]
@ziafazal Kindly review again.

**Pull request merged**